RecoTracker/LSTCore/src/alpaka/MiniDoublet.h

```diff
@@ -0,0 +1,914 @@
+                                                   lowerModuleIndex,
+                                                   upperModuleIndex,
+                                                   lowerHitArrayIndex,
+                                                   upperHitArrayIndex,
+                                                   dz,
+                                                   dphi,
+                                                   dphichange,
+                                                   shiftedX,
+                                                   shiftedY,
+                                                   shiftedZ,
+                                                   noShiftedDphi,
+                                                   noShiftedDphiChange,
+                                                   xLower,
+                                                   yLower,
+                                                   zLower,
+                                                   rtLower,
+                                                   xUpper,
+                                                   yUpper,
+                                                   zUpper,
+                                                   rtUpper);
+          if (success) {
+            int totOccupancyMDs = alpaka::atomicAdd(
+                acc, &mdsOccupancy.totOccupancyMDs()[lowerModuleIndex], 1u, alpaka::hierarchy::Threads{});
+            if (totOccupancyMDs >= (ranges.miniDoubletModuleOccupancy()[lowerModuleIndex])) {
+#ifdef WARNINGS
+              printf("Mini-doublet excess alert! Module index =  %d\n", lowerModuleIndex);
+#endif
+            } else {
+              int mdModuleIndex =
+                  alpaka::atomicAdd(acc, &mdsOccupancy.nMDs()[lowerModuleIndex], 1u, alpaka::hierarchy::Threads{});
+              unsigned int mdIndex = ranges.miniDoubletModuleIndices()[lowerModuleIndex] + mdModuleIndex;
+
+              addMDToMemory(acc,
+                            mds,
+                            hits,
+                            modules,
+                            lowerHitArrayIndex,
+                            upperHitArrayIndex,
+                            lowerModuleIndex,
+                            dz,
+                            dphi,
+                            dphichange,
+                            shiftedX,
+                            shiftedY,
+                            shiftedZ,
+                            noShiftedDphi,
+                            noShiftedDphiChange,
+                            mdIndex);
+            }
+          }
+        }
+      }
+    }
+  };
+
+  struct CreateMDArrayRangesGPU {
+    template <typename TAcc>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc, ModulesConst modules, ObjectRanges ranges) const {
+      // implementation is 1D with a single block
+      static_assert(std::is_same_v<TAcc, ALPAKA_ACCELERATOR_NAMESPACE::Acc1D>, "Should be Acc1D");
```

Comment:
Would simple
```suggestion
    ALPAKA_FN_ACC void operator()(Acc1D const& acc, ModulesConst modules, ObjectRanges ranges) const {
```
work? (can be left for beyond this PR)